src/execution_plan/execution_plan.c

```diff
@@ -683,7 +713,7 @@ void ExecutionPlan_Print(const ExecutionPlan *plan, RedisModuleCtx *ctx) {
 
     // No idea how many operation are in execution plan.
     RedisModule_ReplyWithArray(ctx, REDISMODULE_POSTPONED_ARRAY_LEN);
-    _ExecutionPlan_Print(plan->root, ctx, buffer, 0, &op_count);
+    _ExecutionPlan_Print(plan->root, ctx, buffer, 1024, 0, &op_count);
```

Comment:
Commenting here regarding the entire op->toString chain - it looks to me like all the `snprintf` calls are safely performed currently, but we are now introducing user strings into our buffers, so we need to be very cautious about buffer overflows and should probably switch to extensible-buffer logic in general.

(That said, this change is a great improvement! I like it a lot; it just gets into the area of some classic C vulnerabilities.)

Reply:
Thanks, agree, I don't mind displaying omitted strings if we find ourselves in a situation of buffer overrun.